ods_ci/libs/DataSciencePipelinesAPI.py

```diff
@@ -0,0 +1,242 @@
+import subprocess
+from json import JSONDecodeError
+import requests
+import os
+import json
+import time
+from robotlibcore import keyword
+import base64
+
+
+class DataSciencePipelinesAPI:
+
+    def __init__(self):
+        self.auth_url = self.retrieve_auth_url()
+        self.route = ''
+        self.sa_token = None
+
+    @keyword
+    def create_operator(self, file):
+        print(os.path.abspath(__file__))
+        current_path = __file__.index('/libs/')
+        current_path = __file__[:current_path]
+        response, error = self.run_oc(f"oc apply -f {current_path}/{file}")
+        print(f'create_operator({error}): {response}')
+
+    @keyword
+    def get_redhat_openshift_pipelines(self):
+        """
+        when creating at the first time, it can take like 1 minute to have the pods ready
+        """
+        deployment_count = 0
+        while deployment_count != 1:
+            deployments = []
+            response, _ = self.run_oc(f"oc get deployment -n openshift-operators openshift-pipelines-operator -o json")
+            try:
+                response = json.loads(response)
+                if response['metadata']['name'] == 'openshift-pipelines-operator' and 'readyReplicas' in response['status'] and response['status']['readyReplicas'] == 1:
+                    deployments.append(response)
+            except JSONDecodeError:
+                pass
+            deployment_count = len(deployments)
+            time.sleep(1)
+        return self.count_running_pods(f'oc get pods -n openshift-operators -l name=openshift-pipelines-operator -o json', 'openshift-pipelines-operator', 'Running', 1)
+
+    @keyword
+    def get_data_science_pipelines_operator_controller_manager(self):
```

Comment:
It should be renamed as `Wait Until Data Science Pipelines Controller Manager Is Deployed`
As a suggestion for a future PR, this method could be implemented as regular Robot Framework keyword using the robotframework-openshift library.